R/textplot_scale1d.R

```diff
@@ -135,4 +155,108 @@ textplot_scale1d.textmodel_wordfish_fitted <-
     suppressMessages(p)
 }
 
+#' @noRd
+#' @importFrom stats reorder aggregate
+#' @importFrom ggplot2 ggplot aes geom_point element_blank geom_pointrange 
+#' @importFrom ggplot2 coord_flip xlab ylab theme_bw geom_text theme geom_point
+#' @importFrom ggplot2 facet_grid element_line
+#' @export
+textplot_scale1d.textmodel_wordscores_predicted <- 
+    function(x, margin = c("documents", "features"), 
+             rescaling_plot = "raw", doclabels = NULL, 
+             sort = TRUE, groups = NULL, highlighted = NULL, alpha = 0.7, 
+             highlighted_color = "black") {
+      
+      margin <- match.arg(margin)
+
+      if (margin == "documents") {
+        n <- length(x@textscores$textscore_raw)
+        if (is.null(doclabels))
+          doclabels <- x@newdata@Dimnames$docs
+        stopifnot(length(doclabels) == n)
+
+        if (sort & !is.null(groups)) {
+          temp_medians <- aggregate(x@textscores$textscore_raw, list(groups), median, na.rm = TRUE)
+          groups <- factor(groups, levels = temp_medians[order(temp_medians$x, decreasing = TRUE), 1])
+        }
+        
+        if (rescaling_plot == "raw") {
+          textscore_raw <- textscore_raw_lo <- textscore_raw_hi <- NULL
+          textscore_lbg <- textscore_lbg_lo <- textscore_lbg_hi <- NULL
+          
+          results <- data.frame(doclabels = doclabels, 
+                              textscore = x@textscores$textscore_raw, 
+                              lower = x@textscores$textscore_raw_lo, 
+                              upper = x@textscores$textscore_raw_hi)
+        }
+        
+        if (rescaling_plot == "lbg") {
+          textscore_lbg <- textscore_lbg_lo <- textscore_lbg_hi <- NULL
+          results <- data.frame(doclabels = doclabels, 
+                                textscore = x@textscores$textscore_lbg, 
+                                lower = x@textscores$textscore_lbg_lo, 
+                                upper = x@textscores$textscore_lbg_hi)
+        }
+        
+        if (rescaling_plot == "mv"){
+          textscore_mv <- textscore_mv_lo <- textscore_mv_hi <- NULL
+          results <- data.frame(doclabels = doclabels, 
+                                textscore = x@textscores$textscore_mv)
+        }
+        if (!is.null(groups))
+          results$groups <- groups
+        
+        p <- if (sort) {
+          ggplot(data = results, aes(x = reorder(doclabels, textscore), y = textscore))
+        } else {
+          ggplot(data = results, aes(x = doclabels, y = textscore))
+        }
+        p <- p + 
+          coord_flip() + 
+          { if (!is.null(groups))
+            facet_grid(as.factor(groups) ~ ., scales = "free_y", space = "free") } +  
+          { if (rescaling_plot != "mv")
```

Comment:
Something seems wrong here. In my examples `predict(...rescaling = "mv")` did not produce confidence intervals, so I added this line to avoid an error. But I presume there are better ways around this.